internal/backend/azure/azure.go

```diff
@@ -255,48 +257,46 @@ func (be *Backend) Save(ctx context.Context, h backend.Handle, rd backend.Rewind
 	}
 
 	var err error
-	if rd.Length() < saveLargeSize {
-		// if it's smaller than 256miB, then just create the file directly from the reader
-		err = be.saveSmall(ctx, objName, rd, accessTier)
+	fileSize := rd.Length()
+
+	// If the file size is less than or equal to the max size for a single blob, use the single blob upload
+	// otherwise, use the block-based upload
+	if fileSize <= singleBlobMaxSize {
+		err = be.saveSingleBlob(ctx, objName, rd, accessTier)
 	} else {
-		// otherwise use the more complicated method
 		err = be.saveLarge(ctx, objName, rd, accessTier)
 	}
 
 	return err
 }
 
-func (be *Backend) saveSmall(ctx context.Context, objName string, rd backend.RewindReader, accessTier blob.AccessTier) error {
+// saveSingleBlob uploads data using a single Put Blob operation.
+// This method is more efficient for files under 5000 MiB as it requires only one API call
+// instead of the two calls (StageBlock + CommitBlockList) required by the block-based approach.
+func (be *Backend) saveSingleBlob(ctx context.Context, objName string, rd backend.RewindReader, accessTier blob.AccessTier) error {
 	blockBlobClient := be.container.NewBlockBlobClient(objName)
 
-	// upload it as a new "block", use the base64 hash for the ID
-	id := base64.StdEncoding.EncodeToString(rd.Hash())
-
 	buf := make([]byte, rd.Length())
```

Comment:
We need a way to get rid of this allocation. Keeping multi gigabyte packs in memory is a bad idea.

Reply:
We have to keep it. In some cases tree blobs can become very large and result in pack files that are a few GB in size. I'd also like to bump the maximum pack size to roughly 1GB sometime soon.